Error when coredns resources in helm values file is used

### What happened?

When trying to set resource for coredns in the helm values, `resources` is wrongly used.

All the others vcluster componants can be set using term `resources` with something like

```
syncer:
  image: loftsh/vcluster:0.11.1
  replicas: 3
  resources:
    limits:
      memory: 372Mi
      cpu: 200m
    requests:
      cpu: 50m
      memory: 128Mi
```

Except that coredns does not work as it is intended. When set, it gives the following error:

```
/opt/homebrew/bin/helm --repo=https://charts.loft.sh upgrade -i --reset-values --create-namespace --values=output/myvluster.k8s.values.yaml baslab-infra-vc vcluster-k8s", "msg": "Failure when executing Helm command. Exited 1.\nstdout: \nstderr: coalesce.go:223: warning: destination for vcluster-k8s.coredns.resources is a table. Ignoring non-table value ()\nError: UPGRADE FAILED: template: vcluster-k8s/templates/coredns.yaml:10:38: executing \"vcluster-k8s/templates/coredns.yaml\" at <4>: wrong type for value; expected string; got map[string]interface {}\n", "stderr": "coalesce.go:223: warning: destination for vcluster-k8s.coredns.resources is a table. Ignoring non-table value ()\nError: UPGRADE FAILED: template: vcluster-k8s/templates/coredns.yaml:10:38: executing \"vcluster-k8s/templates/coredns.yaml\" at <4>: wrong type for value; expected string; got map[string]interface {}\n", "stderr_lines": ["coalesce.go:223: warning: destination for vcluster-k8s.coredns.resources is a table. Ignoring non-table value ()", "Error: UPGRADE FAILED: template: vcluster-k8s/templates/coredns.yaml:10:38: executing \"vcluster-k8s/templates/coredns.yaml\" at <4>: wrong type for value; expected string; got map[string]interface {}
```

A quick look at the charts file in `charts/k8s/templates/coredns.yaml` and you can see that it expecting a configmap instead of standard resources.

Can it be changed to `coredns.configmap` and use `coredns.resources` as the others component does ?

`coredns.yaml` could be modified with something like that:

```
  coredns.yaml: |-
          containers:
            - name: coredns
              resources:
    {{ toYaml .Values.coredns.resources | indent 10 }}
          {{- $counter := -1 -}}
          {{- range $key, $container := .Values.plugin }}
          {{- $counter = add1 $counter }}
```

### What did you expect to happen?

When `coredns.resources`  is set, use standard resources values instead of providing configmap

### How can we reproduce it (as minimally and precisely as possible)?

Create helm value with the following

```
# Scale up DNS server
coredns:
  image: ci-docker-dev.artifactory.unil.ch/coredns/coredns:1.8.6
  replicas: 3
  resources:
    requests:
      cpu: 50m
      memory: 256Mi
    limits:
      memory: 824Mi
      cpu: 400m
  securityContext:
    allowPrivilegeEscalation: false
    capabilities:
      drop:
      - ALL
    readOnlyRootFilesystem: true
    runAsGroup: 12345
    runAsNonRoot: true
    runAsUser: 12345
    seccompProfile:
      type: RuntimeDefault
  priorityClassName: high
```
create or upgrade vcluster with those values.

### Anything else we need to know?

Setting 1000m cpu limit for coredns is way to much in my opinion, especially when you enable quota and have 3 replicas of coredns and `limit.cpu` is set to 4000m. Vcluster users would only have 1000m `limit.cpu` left.
If helm charts values can't be changed, at least lower the limit value for coredns.

### Host cluster Kubernetes version

<details>

```console
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.5", GitCommit:"5c99e2ac2ff9a3c549d9ca665e7bc05a3e18f07e", GitTreeState:"clean", BuildDate:"2021-12-16T08:38:33Z", GoVersion:"go1.16.12", Compiler:"gc", Platform:"darwin/arm64"}
Server Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.7", GitCommit:"42c05a547468804b2053ecf60a3bd15560362fc2", GitTreeState:"clean", BuildDate:"2022-05-24T12:24:41Z", GoVersion:"go1.17.10", Compiler:"gc", Platform:"linux/amd64"}
```

</details>


### Host cluster Kubernetes distribution

<details>

```
1.23.7
```

</details>


### vlcuster version

<details>

```console
$ vcluster --version
vcluster version 0.11.1
```

</details>


### Vcluster Kubernetes distribution(k3s(default)), k8s, k0s)

<details>

```
k8s
```

</details>


### OS and Arch

<details>

```
OS: Redhat 9
Arch: amd64
```
</details>

--- a/charts/eks/templates/coredns.yaml
+++ b/charts/eks/templates/coredns.yaml
@@ -4,9 +4,9 @@ metadata:
   name: custom-deployments
   namespace: {{ .Release.Namespace }}
 data:
-{{- if .Values.coredns.resources }}
+{{- if .Values.coredns.manifests }}
   coredns.yaml: |-
-{{ .Values.coredns.resources | indent 4 }}
+{{ .Values.coredns.manifests | indent 4 }}
 {{- else }}
   coredns.yaml: |-
     apiVersion: v1
@@ -146,12 +146,7 @@ data:
             image: "{{ .Values.defaultImageRegistry }}{{ .Values.coredns.image }}"
             imagePullPolicy: IfNotPresent
             resources:
-              limits:
-                cpu: 1000m
-                memory: 170Mi
-              requests:
-                cpu: 3m
-                memory: 16Mi
+{{ toYaml .Values.coredns.resources | indent 14}}
             args: [ "-conf", "/etc/coredns/Corefile" ]
             volumeMounts:
             - name: config-volume

--- a/charts/eks/values.yaml
+++ b/charts/eks/values.yaml
@@ -211,7 +211,18 @@ api:
 coredns:
   image: public.ecr.aws/eks-distro/coredns/coredns:v1.8.7-eks-1-23-4
   replicas: 1
-  resources: ""
+  resources:
+    limits:
+      cpu: 1000m
+      memory: 170Mi
+    requests:
+      cpu: 3m
+      memory: 16Mi
+# if below option is configured, it will override the coredns manifests with the following string
+#  manifests: |-
+#    apiVersion: ...
+#    ...
+
 # Service account that should be used by the vcluster
 serviceAccount:
   create: true

--- a/charts/k0s/templates/coredns.yaml
+++ b/charts/k0s/templates/coredns.yaml
@@ -5,9 +5,9 @@ metadata:
   name: {{ .Release.Name }}-coredns
   namespace: {{ .Release.Namespace }}
 data:
-{{- if .Values.coredns.allresources }}
+{{- if .Values.coredns.manifests }}
   coredns.yaml: |-
-{{ .Values.coredns.allresources | indent 4 }}
+{{ .Values.coredns.manifests | indent 4 }}
 {{- else }}
   coredns.yaml: |-
     apiVersion: v1

--- a/charts/k0s/templates/coredns.yaml
+++ b/charts/k0s/templates/coredns.yaml
@@ -5,9 +5,9 @@ metadata:
   name: {{ .Release.Name }}-coredns
   namespace: {{ .Release.Namespace }}
 data:
-{{- if .Values.coredns.resources }}
+{{- if .Values.coredns.allresources }}
   coredns.yaml: |-
-{{ .Values.coredns.resources | indent 4 }}
+{{ .Values.coredns.allresources | indent 4 }}
 {{- else }}
   coredns.yaml: |-
     apiVersion: v1
@@ -147,12 +147,7 @@ data:
               {{- end }}
               imagePullPolicy: IfNotPresent
               resources:
-                limits:
-                  cpu: 1000m
-                  memory: 170Mi
-                requests:
-                  cpu: 3m
-                  memory: 16Mi
+{{ toYaml .Values.coredns.resources | indent 16}}
               args: [ "-conf", "/etc/coredns/Corefile" ]
               volumeMounts:
                 - name: config-volume

--- a/charts/k0s/values.yaml
+++ b/charts/k0s/values.yaml
@@ -268,7 +268,15 @@ coredns:
     externalTrafficPolicy: ""
     # Extra Annotations
     annotations: {}
-  resources: ""
+  resources:
+    requests:
+      cpu: 50m
+      memory: 256Mi
+    limits:
+      memory: 24Mi
+      cpu: 400m
+# if the below is specified, the above resources will not be entertained. Below key is in string format.
+#  allresources: |
 
 # If enabled will deploy vcluster in an isolated mode with pod security
 # standards, limit ranges and resource quotas

--- a/charts/k0s/values.yaml
+++ b/charts/k0s/values.yaml
@@ -269,15 +269,16 @@ coredns:
     # Extra Annotations
     annotations: {}
   resources:
-    requests:
-      cpu: 50m
-      memory: 256Mi
     limits:
-      memory: 24Mi
-      cpu: 400m
-# if the below is specified, the above resources will not be entertained. Below key is in string format.
-#  allresources: |
-
+      cpu: 1000m
+      memory: 170Mi
+    requests:
+      cpu: 3m
+      memory: 16Mi
+# if below option is configured, it will override the coredns manifests with the following string
+#  manifests: |-
+#    apiVersion: ...
+#    ...
 # If enabled will deploy vcluster in an isolated mode with pod security
 # standards, limit ranges and resource quotas
 isolation:

--- a/charts/k3s/templates/coredns.yaml
+++ b/charts/k3s/templates/coredns.yaml
@@ -5,9 +5,9 @@ metadata:
   name: {{ .Release.Name }}-coredns
   namespace: {{ .Release.Namespace }}
 data:
-{{- if .Values.coredns.allresources }}
+{{- if .Values.coredns.manifests }}
   coredns.yaml: |-
-{{ .Values.coredns.allresources | indent 4 }}
+{{ .Values.coredns.manifests | indent 4 }}
 {{- else }}
   coredns.yaml: |-
     apiVersion: v1

--- a/charts/k3s/templates/coredns.yaml
+++ b/charts/k3s/templates/coredns.yaml
@@ -5,9 +5,9 @@ metadata:
   name: {{ .Release.Name }}-coredns
   namespace: {{ .Release.Namespace }}
 data:
-{{- if .Values.coredns.resources }}
+{{- if .Values.coredns.allresources }}
   coredns.yaml: |-
-{{ .Values.coredns.resources | indent 4 }}
+{{ .Values.coredns.allresources | indent 4 }}
 {{- else }}
   coredns.yaml: |-
     apiVersion: v1
@@ -147,12 +147,7 @@ data:
               {{- end }}
               imagePullPolicy: IfNotPresent
               resources:
-                limits:
-                  cpu: 1000m
-                  memory: 170Mi
-                requests:
-                  cpu: 3m
-                  memory: 16Mi
+{{ toYaml .Values.coredns.resources | indent 16}}
               args: [ "-conf", "/etc/coredns/Corefile" ]
               volumeMounts:
                 - name: config-volume

--- a/charts/k3s/values.yaml
+++ b/charts/k3s/values.yaml
@@ -270,15 +270,16 @@ coredns:
     # Extra Annotations
     annotations: {}
   resources:
-    requests:
-      cpu: 50m
-      memory: 256Mi
     limits:
-      memory: 824Mi
-      cpu: 400m
-# if the below is specified, the above resources will not be entertained. Below key is in string format.
-#  allresources: |
-
+      cpu: 1000m
+      memory: 170Mi
+    requests:
+      cpu: 3m
+      memory: 16Mi
+# if below option is configured, it will override the coredns manifests with the following string
+#  manifests: |-
+#    apiVersion: ...
+#    ...
 # If enabled will deploy vcluster in an isolated mode with pod security
 # standards, limit ranges and resource quotas
 isolation:

--- a/charts/k3s/values.yaml
+++ b/charts/k3s/values.yaml
@@ -269,7 +269,15 @@ coredns:
     externalTrafficPolicy: ""
     # Extra Annotations
     annotations: {}
-  resources: ""
+  resources:
+    requests:
+      cpu: 50m
+      memory: 256Mi
+    limits:
+      memory: 824Mi
+      cpu: 400m
+# if the below is specified, the above resources will not be entertained. Below key is in string format.
+#  allresources: |
 
 # If enabled will deploy vcluster in an isolated mode with pod security
 # standards, limit ranges and resource quotas

--- a/charts/k8s/templates/coredns.yaml
+++ b/charts/k8s/templates/coredns.yaml
@@ -5,9 +5,9 @@ metadata:
   name: {{ .Release.Name }}-coredns
   namespace: {{ .Release.Namespace }}
 data:
-{{- if .Values.coredns.allresources }}
+{{- if .Values.coredns.manifests }}
   coredns.yaml: |-
-{{ .Values.coredns.allresources | indent 4 }}
+{{ .Values.coredns.manifests | indent 4 }}
 {{- else }}
   coredns.yaml: |-
     apiVersion: v1

--- a/charts/k8s/templates/coredns.yaml
+++ b/charts/k8s/templates/coredns.yaml
@@ -5,9 +5,9 @@ metadata:
   name: {{ .Release.Name }}-coredns
   namespace: {{ .Release.Namespace }}
 data:
-{{- if .Values.coredns.resources }}
+{{- if .Values.coredns.allresources }}
   coredns.yaml: |-
-{{ .Values.coredns.resources | indent 4 }}
+{{ .Values.coredns.allresources | indent 4 }}
 {{- else }}
   coredns.yaml: |-
     apiVersion: v1
@@ -147,12 +147,7 @@ data:
               {{- end }}
               imagePullPolicy: IfNotPresent
               resources:
-                limits:
-                  cpu: 1000m
-                  memory: 170Mi
-                requests:
-                  cpu: 3m
-                  memory: 16Mi
+{{ toYaml .Values.coredns.resources | indent 16}}
               args: [ "-conf", "/etc/coredns/Corefile" ]
               volumeMounts:
                 - name: config-volume

--- a/charts/k8s/values.yaml
+++ b/charts/k8s/values.yaml
@@ -307,7 +307,16 @@ coredns:
     externalTrafficPolicy: ""
     # Extra Annotations
     annotations: {}
-  resources: ""
+  resources:
+    requests:
+      cpu: 50m
+      memory: 256Mi
+    limits:
+      memory: 824Mi
+      cpu: 400m
+# if the below is specified, the above resources will not be entertained. Below key is in string format.
+#  allresources: |
+
 
 # If enabled will deploy vcluster in an isolated mode with pod security
 # standards, limit ranges and resource quotas

--- a/charts/k8s/values.yaml
+++ b/charts/k8s/values.yaml
@@ -308,15 +308,16 @@ coredns:
     # Extra Annotations
     annotations: {}
   resources:
-    requests:
-      cpu: 50m
-      memory: 256Mi
     limits:
-      memory: 824Mi
-      cpu: 400m
-# if the below is specified, the above resources will not be entertained. Below key is in string format.
-#  allresources: |
-
+      cpu: 1000m
+      memory: 170Mi
+    requests:
+      cpu: 3m
+      memory: 16Mi
+# if below option is configured, it will override the coredns manifests with the following string
+#  manifests: |-
+#    apiVersion: ...
+#    ...
 
 # If enabled will deploy vcluster in an isolated mode with pod security
 # standards, limit ranges and resource quotas

